generic_multisig.py

```diff
@@ -74,30 +53,50 @@ def execute(self, execution_request):
             sp.if sp.check_signature(operator_public_key, execution_request.signatures[sp.hash_key(operator_public_key)], sp.pack(signing_payload)):
                 valid_signatures_counter.value += 1
         sp.verify(valid_signatures_counter.value >= self.data.signers_threshold)
+        self.data.nonce += 1
+        sp.add_operations(execution_request.execution_payload(sp.unit).rev())
+        
 
-        # Optionally timelock the request.
-        sp.if self.data.timelock_seconds.is_some():
-            self.data.timelock[self.data.nonce] = (sp.now, execution_request)
-        sp.else:
-            sp.add_operations(execution_request.execution_payload(sp.unit).rev())
 
-        self.data.nonce += 1
+class TimelockedExecutionRequest():
+    def get_type():
+        return sp.TRecord(creation_timestamp=sp.TTimestamp, execution_request=ExecutionRequest.get_type()).layout(("creation_timestamp", "execution_request"))
 
-    # Param: A nat representing the nonce for the proposal to execute. 
-    @sp.entry_point
-    def executeTimelock(self, nonce):
-        sp.set_type(nonce, sp.TNat)
 
-        # Get timelock. Will fail if there's no request for nonce.
-        timelock_value = self.data.timelock[nonce]
-        timelock_start, execution_request = sp.match_pair(timelock_value)
+class TimeLockedExecutor(Executor):
+    def get_init_storage(self):
+        storage = super().get_init_storage()
```

Comment:
I think you can pass args up directly using `**extra_storage` as a param to the initializer in the super class. Any reason you don't do that?